beacon-chain/operations/blstoexec/pool.go

```diff
@@ -0,0 +1,130 @@
+package blstoexec
+
+import (
+	"math"
+	"sync"
+
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+)
+
+type listNode struct {
+	value *ethpb.SignedBLSToExecutionChange
+	prev  *listNode
+	next  *listNode
+}
+
+type changesList struct {
+	first *listNode
+	last  *listNode
+	len   int
+}
+
+// PoolManager maintains pending and seen BLS-to-execution-change objects.
+// This pool is used by proposers to insert BLS-to-execution-change objects into new blocks.
+type PoolManager interface {
+	PendingBLSToExecChanges(returnAll bool) []*ethpb.SignedBLSToExecutionChange
+	InsertBLSToExecChange(change *ethpb.SignedBLSToExecutionChange)
+	MarkIncluded(change *ethpb.SignedBLSToExecutionChange)
+}
+
+// Pool is a concrete implementation of PoolManager.
+type Pool struct {
+	lock    sync.RWMutex
+	pending changesList
+	m       map[types.ValidatorIndex]*listNode
+}
+
+// NewPool returns an initialized pool.
+func NewPool() *Pool {
+	return &Pool{
+		pending: changesList{},
+		m:       make(map[types.ValidatorIndex]*listNode),
+	}
+}
+
+// PendingBLSToExecChanges returns objects that are ready for inclusion at the given slot.
+// Without returnAll, this method will not return more than the block enforced MaxBlsToExecutionChanges.
+func (p *Pool) PendingBLSToExecChanges(returnAll bool) []*ethpb.SignedBLSToExecutionChange {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	if returnAll {
```

Comment:
This looks like a bool param that DeepSource will complain, probably better to have two different getters.